src/pages/inventory/index.tsx

```diff
@@ -0,0 +1,265 @@
+  const indexOfFirstItem = indexOfLastItem - itemsPerPage;
+  const currentItems = data.slice(indexOfFirstItem, indexOfLastItem);
+
+  const handleTypeClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorType(event.currentTarget);
+  };
+  const handleCategoryClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorCategory(event.currentTarget);
+  };
+  const handleStatusClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorStatus(event.currentTarget);
+  };
+
+  const handleTypeClose = () => {
+    setAnchorType(null);
+  }
+
+  const handleMenuTypeClick = (value: string) => {
+    setType(value)
+    handleFilter();
+    handleTypeClose();
+  }
+
+  const handleCategoryClose = () => {
+    setAnchorCategory(null);
+  }
+
+  const handleMenuCategoryClick = (value: string) => {
+    setCategory(value);
+    handleFilter();
+    handleCategoryClose();
+  }
+
+  const handleStatusClose = () => {
+    setAnchorStatus(null);
+  }
+
+  const handleMenuStatusClick = (value: string) => {
+    setStatus(value);
+    handleFilter();
+    handleStatusClose();
+  }
+
+  const clearTypeFilter = () => {
+    setType('');
+  }
+  const clearCategoryFilter = () => {
+    setCategory('');
+  }
+  const clearStatusFilter = () => {
+    setStatus('');
+  }
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearch(event.target.value)
+  }
+
+  const handlePageChange = (_event: React.ChangeEvent<unknown>, value: number) => {
+    setCurrentPage(value);
+  }
+
+  const handleFilter = () => {
+    const searchFiltered = originalData.filter((row: { item: string, type: string, category: string, quantity: number; }) => {
+
+      const matchesType = type ? row.type.toLowerCase().includes(type.toLowerCase()) : true;
+
+      const matchesCategory = category ? row.category.toLowerCase().includes(category.toLowerCase()) : true;
+
+      // const matchesStatus = status ? item.status.toLowerCase().includes(status.toLowerCase()) : true; //
+
+      const matchesSearch = search
+        ? row.item.toLowerCase().includes(search.toLowerCase()) ||
+        row.type.toLowerCase().includes(search.toLowerCase()) ||
+        row.category.toLowerCase().includes(search.toLowerCase()) ||
+        // item.status.toLowerCase().includes(search.toLowerCase()) ||
+        row.quantity.toString().toLowerCase().includes(search.toLowerCase())
+        : true;
+
+      return matchesType && matchesCategory && matchesSearch;
+    });
+
+    setData(searchFiltered);
+  };
+
+  const supabase = createClient(import.meta.env.VITE_SUPABASE_URL, import.meta.env.VITE_SUPABASE_ANON_KEY); // When we deploy this application or when we create a server, this will have to be changed to server side so that the anon_key can be hidden.
+
+  const fetchData = async () => {
+    const { data: inventory, error } = await supabase
+      .from('inventory')
+      .select(`id, item, type, quantity, category`);
```

Comment:
Front-end pagination looks good, but given the potential increase in data volume in the future, it would be better to have pagination logic built into the database. Supabase already supports fetching data by range

```suggestion
  const fetchData = async () => {
    const { data: inventory, error } = await supabase
      .from('inventory')
      .select(`id, item, type, quantity, category`)
      .range(startIndex, endIndex); //add logic for correct pagination
      <!-- 
        For example
        const itemsPerPage = 10;
        const currentPage = 1; // this can be dynamic
        const startIndex = (currentPage - 1) * itemsPerPage;
        const endIndex = startIndex + itemsPerPage - 1;
        -->

```

Reply:
This will be updated when we move away from supabase. I'm going to leave as is for now so we can merge.

Reply:
agreed